### PR TITLE
Introduce a concept of Command Descriptions for TF-IDF indexing

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
COMPETING PR TO https://github.com/microsoft/vscode/pull/193631

This allows workbench actions to contribute ILocalizedStrings as description (or they can still use the complex `ICommandHandlerDescription` type).

If a description is contributed, then that will be included in the TF-IDF chunk calculation. This allows us to include additional relevant information that will help the Command Palette find the command you're looking for.

image
image

Additionally, these descriptions are diplayed in the keybindings.json editing experience:

image
 

#### Next steps

After this goes in, I'll work on lighting up the same for Editor Commands.

### CODE REVIEW

File: src/vs/workbench/services/keybinding/browser/keybindingService.ts [hunk 1]
```diff
@@ -906,13 +907,18 @@ class KeybindingsJsonSchema {
 		this.commandsEnumDescriptions.length = 0;
 
 		const knownCommands = new Set<string>();
-		const addKnownCommand = (commandId: string, description?: string | undefined) => {
+		const addKnownCommand = (commandId: string, description?: ILocalizedString | ICommandHandlerDescription | undefined) => {
```
Comment (TylerLeonhardt) [inline src/vs/workbench/services/keybinding/browser/keybindingService.ts:910]: @ulugbekna I saw your name in the blame of these changes. Can you review from your perspective here?

File: src/vs/workbench/api/browser/mainThreadCommands.ts [hunk 2]
```diff
@@ -98,9 +99,12 @@ export class MainThreadCommands implements MainThreadCommandsShape {
 
 // --- command doc
 
-function _generateMarkdown(description: string | Dto<ICommandHandlerDescription> | ICommandHandlerDescription): string {
+function _generateMarkdown(description: string | ILocalizedString | Dto<ICommandHandlerDescription> | ICommandHandlerDescription): string {
 	if (typeof description === 'string') {
 		return description;
+	} else if (isLocalizedString(description)) {
```
Comment (TylerLeonhardt) [inline src/vs/workbench/api/browser/mainThreadCommands.ts:105]: @gregvanl said it is ok to include these descriptions in the output of this command.

File: src/vs/workbench/contrib/files/browser/fileActions.contribution.ts [hunk 3]
```diff
@@ -189,29 +189,89 @@ function appendSaveConflictEditorTitleAction(id: string, title: string, icon: Th
...
+	id: NEW_FILE_COMMAND_ID,
+	title: { value: NEW_FILE_LABEL, original: 'New File' },
+	category: Categories.File
+}, WorkspaceFolderCountContext.notEqualsTo('0'));
+
+appendToCommandPalette({
+	id: NEW_FOLDER_COMMAND_ID,
+	title: { value: NEW_FOLDER_LABEL, original: 'New Folder' },
+	category: Categories.File,
+	description: { value: nls.localize('newFolderDescription', "Create a new folder or directory"), original: 'Create a new folder or directory' }
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/files/browser/fileActions.contribution.ts:267]: This is one of the examples

File: src/vs/workbench/contrib/localization/common/localizationsActions.ts [hunk 4]
```diff
@@ -23,6 +23,10 @@ export class ConfigureDisplayLanguageAction extends Action2 {
 			title: { original: 'Configure Display Language', value: ConfigureDisplayLanguageAction.LABEL },
 			menu: {
 				id: MenuId.CommandPalette
+			},
+			description: {
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/localization/common/localizationsActions.ts:27]: This is one of the examples.

File: src/vs/workbench/services/keybinding/browser/keybindingService.ts (hunk 1 above)
Comment (ulugbekna) [reply src/vs/workbench/services/keybinding/browser/keybindingService.ts:910]: Looks good to me.

**Review by ulugbekna — APPROVED** [review]

**Pull request closed without merging**